uniref-rest/src/test/java/org/uniprot/api/uniref/repository/store/UniRefEntryStoreRepositoryTest.java

```diff
@@ -0,0 +1,367 @@
+        assertNotNull(result.getPage());
+        CursorPage page = result.getPage();
+        assertEquals(20L, page.getOffset().longValue());
+        assertEquals(10, page.getPageSize().intValue());
+        assertEquals(28L, page.getTotalElements().longValue());
+        assertFalse(page.hasNextPage());
+
+        assertNotNull(result.getEntry());
+
+        UniRefEntry entry = result.getEntry();
+        assertNotNull(entry);
+        assertNull(entry.getId());
+        assertNull(entry.getName());
+        assertNull(entry.getEntryType());
+        assertNull(entry.getUpdated());
+        assertNull(entry.getCommonTaxon());
+        assertNull(entry.getCommonTaxonId());
+        assertTrue(entry.getGoTerms().isEmpty());
+        assertNull(entry.getRepresentativeMember());
+        assertNull(entry.getMemberCount());
+        assertEquals(8, entry.getMembers().size());
+    }
+
+    @Test
+    void getNotFoundEntryLight() {
+        UniRefIdRequest request = new UniRefIdRequest();
+        ResourceNotFoundException error =
+                assertThrows(
+                        ResourceNotFoundException.class,
+                        () -> repository.getEntryById("NOT_FOUND_ID", request));
+        assertEquals(
+                "Unable to get UniRefEntry from store. ClusterId: NOT_FOUND_ID",
+                error.getMessage());
+    }
+
+    @Test
+    void getUniRef90ById() {
+        UniRefIdRequest request = new UniRefIdRequest();
+        UniRefEntryResult result = repository.getEntryById(UNIREF_90_ID, request);
+        assertNotNull(result);
+
+        UniRefEntry entry = result.getEntry();
+        assertEquals(UNIREF_90_ID, entry.getId().getValue());
+
+        assertEquals(2, entry.getMemberCount().intValue());
+        assertEquals(1, entry.getMembers().size());
+
+        assertEquals(REPRESENTATIVE_ID, entry.getRepresentativeMember().getMemberId());
+        assertNull(entry.getRepresentativeMember().isSeed());
+        assertNull(entry.getRepresentativeMember().getUniRef90Id());
+
+        UniRefMember seedMember = entry.getMembers().get(0);
+        assertNotNull(seedMember);
+
+        assertEquals(SEED_ID, seedMember.getMemberId());
+        assertTrue(seedMember.isSeed());
+        assertNull(seedMember.getUniRef90Id());
+    }
+
+    @Test
+    void getRepresentativeSeedEntryById() {
+        UniRefIdRequest request = new UniRefIdRequest();
+        UniRefEntryResult result =
+                repository.getEntryById(UNIREF_REPRESENTATIVE_AS_SEED_ID, request);
+        assertNotNull(result);
+
+        UniRefEntry entry = result.getEntry();
+        assertEquals(UNIREF_REPRESENTATIVE_AS_SEED_ID, entry.getId().getValue());
+
+        assertEquals(1, entry.getMemberCount().intValue());
+        assertEquals(0, entry.getMembers().size());
+
+        assertEquals(SEED_ID, entry.getRepresentativeMember().getMemberId());
+        assertEquals(true, entry.getRepresentativeMember().isSeed());
+    }
+
```

Comment:
Can you please add few negative test cases around size?
e.g. 
1. lets say a cluster has only 10 members but the client is passing size as 100.
2. A cluster has only a rep member and user is passing size as more than 1.

Reply:
1. covered by getUniRef90ById, we have a cluster just with 2 members (1 rep and one member) and default page size is 25 ( I added one assert to make sure that page is null)
2. covered by pages test (last page).